### ISSUE TITLE
scene.enabled() only firing first time

### ISSUE THREAD

**Issue by BenjoJ** [janpaepke/ScrollMagic#283]
Hi Jan

I tried to do a jsfiddle for this but it is just too complicated to replicate properly. I do not have a public facing domain for you to test this on either unfortunately.

I use ScrollMagic to pin #Screen2 and animate the sliding in of divs as the user scrolls:

```html
			<div id="Screen2" class="Screen">
					<div img="/images/temp.jpg" class="Slide" id="img1"></div>
					<div img="/images/temp.jpg" class="Slide" id="img2"></div>
					<div img="/images/temp.jpg" class="Slide" id="img3"></div>
					<div img="/images/temp.jpg" class="Slide" id="img4"></div>
					<div img="/images/temp.jpg" class="Slide" id="img5"></div>
					<div img="/images/temp.jpg" class="Slide" id="img6"></div>
					<div img="/images/temp.jpg" class="Slide" id="img7"></div>
			</div>
```

```js
tween = new TimelineMax()
				   .add(function () { iScreen = 2 })
				   .add([TweenMax.fromTo("#img2", 2, { marginLeft: '100%' }, { marginLeft: 0, ease: Linear.easeNone })])
				   .add([TweenMax.fromTo("#img3", 2, { marginLeft: '100%' }, { marginLeft: 0, ease: Linear.easeNone })])
				   .add([TweenMax.fromTo("#img4", 2, { marginLeft: '100%' }, { marginLeft: 0, ease: Linear.easeNone })])
				   .add([TweenMax.fromTo("#img5", 2, { marginLeft: '100%' }, { marginLeft: 0, ease: Linear.easeNone })])
				   .add([TweenMax.fromTo("#img6", 2, { marginLeft: '100%' }, { marginLeft: 0, ease: Linear.easeNone })])
				   .add([TweenMax.fromTo("#img7", 2, { marginLeft: '100%' }, { marginLeft: 0, ease: Linear.easeNone })])
		PhilosophyScene = new ScrollMagic.Scene({ triggerElement: "#Screen2", duration: 5000, offset: iScreenSize_H / 2 })
						.setTween(tween)
						.setPin("#Screen2")
						.addTo(controller)
```

I have a function that scrolls to an element so the user can jump to a screen:

```js
function scrollToEl(sID) {
        if (jQuery(sID).length) {
                PhilosophyScene.enabled(false);
                var offset = jQuery(sID).offset();
                jQuery("html,body").animate({
                        scrollTop: offset.top - 0,
                        scrollLeft: offset.left
                }, 1000, 'easeOutQuart', function () { PhilosophyScene.enabled(true) });
        }
}
```

You can see here that I am disabling the scene, so that the animation of the sliding-in elements does not occur when jumping to a screen. This works fine, and then the scene is enabled when the jump has completed.

The problem is, when I then hit the function again to go to another element, the "PhilosophyScene.enabled(false);" does not seem to work the second time, causing the animation to run through and the scrolling to then complete after it has finished. 

I hope that makes sense - let me know if you need anymore info.

Thanks

Ben

**Comment by janpaepke** [comment]
If what you're describing is true and replicable it might be a bug.
But let's make sure first it does exactly what you describe.
You are aware, that your scroll function you only disable the PhilosophyScene, right?
So when you're saying "you hit it again and go to another element" is the philosophy scene even still visible or is it maybe another scene, that you didn't disable?

**Comment by BenjoJ** [comment]
Yes the philosophyscene is the scene with the pinned #Scene2 and the animation tween. This is the one I am disabling so that the function can scroll straight past it without the tween running through the animation. 

Just doing some more debugging for you as I have found the situation whereby it does it every time.

**Comment by BenjoJ** [comment]
OK I've sorted it. The problem was occurring when the scrolling animation is called more than once in quick succession, meaning that PhilosophyScene.enabled(false) is being called more than once, then PhilosophyScene.enabled(true) is again being many times upon completion of each animation. 

If I ran the function lots of times, scrolling up past #Screen2 and back down past it again, the scene would not be disabled when scrolling past. If I then left it for 5 seconds and called the function again (once) then it would work as expected, and disable the scene as it scrolled past. 

To fix this in my code I added jQuery("html,body").finish(); to the top of my function, to eliminate the animation should the user double click on any element that calls the function. 

I'm not sure if it is a bug in ScrollMagic, or just a random quirk from my implementation, but hopefully my findings will give you an idea. I hope this all makes sense.

Thanks for your help

Ben
